OpenChange/MAPIStoreMessage.m

```diff
@@ -178,56 +178,109 @@ - (NSDictionary *) _convertRecipient: (struct mapistore_message_recipient *) rec
                           andColumns: (struct SPropTagArray *) columns
 {
   NSMutableDictionary *recipientProperties;
+  enum MAPITAGS prop_tag;
+  char *displayName = NULL;
+  char *email = NULL;
+  uint8_t object_type = 0;
+  char *smtpAddress = NULL;
   SOGoUser *recipientUser;
   NSUInteger count;
+  id recipientEmail = nil;
+  id recipientFullName = nil;
   id value;
 
   recipientProperties = [NSMutableDictionary dictionaryWithCapacity: columns->cValues + 2];
 
+  for (count = 0; count < columns->cValues; count++)
+    {
+      prop_tag = columns->aulPropTag[count];
+      switch(prop_tag) {
+      case PidTagDisplayName:   
+        displayName = recipient->data[count];  
+        break;
+      case PidTagEmailAddress:
+        email = recipient->data[count];
+        break;
+      case PidTagObjectType:
+        object_type = *((uint8_t*) recipient->data[count]);
+        break;
+      case PidTagSmtpAddress:
+        smtpAddress = recipient->data[count];
+        break;
+      default:
+        break;
+      }
+
+      if (recipient->data[count])
+        {
+          value = NSObjectFromValuePointer (prop_tag,
+                                            recipient->data[count]);
+          if (value) 
+            {
+              [recipientProperties setObject: value
+                                      forKey: MAPIPropertyKey (prop_tag)];
+            }       
+        }
+    }
+
   if (recipient->username)
     {
       value = [NSString stringWithUTF8String: recipient->username];
       [recipientProperties setObject: value forKey: @"x500dn"];
+    }
 
-      recipientUser = [SOGoUser userWithLogin: [value lowercaseString]];
-      if (recipientUser)
+  if (object_type == MAPI_MAILUSER && (recipient->username)) {
+    /* values from user object have priority */
+    recipientUser = [SOGoUser userWithLogin: [value lowercaseString]];
+    if (recipientUser)
+      {
+        value = [recipientUser cn];
+        if ([value length] > 0)
+          {
+            recipientFullName = value;
+          }
+        value = [[recipientUser allEmails] objectAtIndex: 0];
```

Comment:
I have issue with getting only the first address, maybe it could get a bad recipient field?. I left it to be conservative; and when this pr is fixed and will check this issue